lib/rules/no-dynamic-id-classes.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * @fileoverview Don't use classes as an identifier
+ * @author Sandeep Baldawa
+ */
+'use strict'
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Use data-* attributes to provide context to your selectors and insulate them from CSS or JS changes https://docs.cypress.io/guides/references/best-practices.html#Selecting-Elements',
+      category: 'Possible Errors',
+      recommended: true,
+      url: 'https://docs.cypress.io/guides/references/best-practices.html#Selecting-Elements',
+    },
+    schema: [],
+    messages: {
+      unexpected: 'Do not use dynamic ids or classes for elements',
+    },
+  },
+
+  create (context) {
+    return {
+      CallExpression (node) {
+        if (isCallingCyGet(node) && isClassName(node)) {
```

Comment:
Instead of blacklisting everything that's not a data attribute like classes, it would be simpler to flip the logic and ensure it *is* a data attribute, then fail if it's not.

Reply:
`cy.get("button")` is marked as a valid selector for this rule though

Reply:
That probably shouldn't be the case though, right?

Reply:
I guess, that is what our best practices say

Reply:
In this rule I was trying to check for just class names, if class names are using as selectors it's an invalid use case. Was thinking to add another rule for data attributes, this way the rules are smaller and easily configurable. I can go either way 🙇

Reply:
Also `cy.get("button")` is a bad example I should probably remove from the units test cases

Reply:
Yah best practices seem to recommend not to use a class name for selectors ```Oftentimes we see users run into problems targeting their elements because:

Your application may use dynamic classes or ID’s that change
Your selectors break from development changes to CSS styles or JS behavior
Luckily, it is very easy to avoid both of these problems.

Don’t target elements based on CSS attributes such as: id, class, tag```

Reply:
Was thinking to have a separate rule for id, class and tag. Let know what you folks think

Reply:
Have addressed the comments, checking if a `data attribute` does not exist, if so error out